src/org/exist/storage/NativeValueIndex.java

```diff
@@ -851,6 +852,7 @@ public NodeSet matchAll(final XQueryWatchDog watchDog, final DocumentSet docs, f
             final int collectionId = iter.next().getId();
 
             watchDog.proceed(null);
+            watchDog.proceed(null);
```

Comment:
@shabanovd Oops! No idea how that happened, I just fixed, rebased and pushed again :-)